src/main/java/Helper/DateHandler.java

```diff
@@ -0,0 +1,190 @@
+
+        for (int i = 0; i < strings.length / 2; i++) {
+            String temp = strings[i];
+            strings[i] = strings[strings.length - i - 1];
+            strings[strings.length - i - 1] = temp;
+        }
+
+        int year_idx = 0;
+        int month_idx = 1;
+        int day_idx = 2;
+        this.date = LocalDate.of(Integer.valueOf(strings[year_idx]), Month.valueOf(strings[month_idx]),
+                Integer.valueOf(strings[day_idx]));
+
+        int splitArrLenWithTime = 5;
+        if (strings.length == splitArrLenWithTime) {
+            //extract time
+            String[] time = strings[strings.length - 2].split(":");
+            int hour_idx = 0;
+            int minute_idx = 1;
+            int hour = Integer.valueOf(time[hour_idx]);
+            int minute = Integer.valueOf(time[minute_idx].substring(1));
+            this.time = LocalTime.of(hour, minute);
+            this.isTimeGiven = true;
+        } else {
+            this.time = LocalTime.MIN;
+            this.isTimeGiven = false;
+        }
+    }
+
+    /**
+     * checks if the userDate is of the valid format.
+     * @param userDate the date input by user.
+     * @throw WrongDateArgumentException if the user inputs an invalid date.
+     * @throw WrongTimeArgumentException if the user inputs an invalid time.
+     */
+    public static void checkValidDate(String userDate) {
+
+        // handle time checking
+        if (userDate.contains(SPACE)) {
+            String hr_mt = userDate.split(SPACE, LIMIT)[TIME_IDX].trim();
+
+            if (hr_mt.length() != TIME_LENGTH) {
+                throw new WrongTimeArgumentException(userDate);
+            }
+
+            for (int i = 0; i < TIME_LENGTH; i++) {
+                if (!Character.isDigit(hr_mt.charAt(i))) {
+                    throw new WrongTimeArgumentException(userDate);
+                }
+            }
+
+            userDate = userDate.substring(START_INDEX, DATE_LENGTH).trim();
+        }
+
+        // handles date checking.
+        char separator = "-".charAt(START_INDEX);
+        int firstSeparatorIdx = 4;
+        int secondSeparatorIdx = 7;
+
+        // checks if the date is in the correct format.
+        if (userDate.length() != DATE_LENGTH) {
+            throw new WrongDateArgumentException(userDate);
+        }
+
+        if (userDate.charAt(firstSeparatorIdx) != separator || userDate.charAt(secondSeparatorIdx) != separator) {
+            throw new WrongDateArgumentException(userDate);
+        }
+
+        for (int i = 0; i < DATE_LENGTH; i++) {
+            if (i != firstSeparatorIdx && i != secondSeparatorIdx) {
+                if (!Character.isDigit(userDate.charAt(i))) {
+                    throw new WrongDateArgumentException(userDate);
+
+                }
+            }
+        }
+    }
+
+    /**
+     * checks whether the date is before the given date.
+     * @param maxDate the value if the date limit
+     * @return true if the true; false otherwise
```

Comment:
I believe you missed out periods at the end of the line here

Reply:
Yup fixed it! 👍